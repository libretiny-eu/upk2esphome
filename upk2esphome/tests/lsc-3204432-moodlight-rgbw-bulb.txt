{
	"manufacturer": "LSC",
	"name": "3204432 Moodlight RGBW Bulb",
	"key": "keydhn7587yertgt",
	"ap_ssid": "SmartLife",
	"github_issues": [],
	"image_urls": [
		"https://tuya-cloudcutter.github.io/images/lsc-3204432-moodlight-rgbw-bulb.jpg"
	],
	"profiles": [
		{
			"slug": "oem-bk7231n-atmosphere-light-ty-2.0.0-sdk-2.3.1-40.00",
			"name": "2.0.0 - BK7231N",
			"type": "CLASSIC",
			"sub_name": "oem_bk7231n_atmosphere_light_ty",
			"icon": "lightbulb-outline"
		}
	],
	"schemas": {
		"eu4aww": [
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"type": "bool"
				},
				"id": 20
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"range": [
						"white",
						"colour",
						"scene",
						"music"
					],
					"type": "enum"
				},
				"id": 21
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"min": 10,
					"max": 1000,
					"scale": 0,
					"step": 1,
					"type": "value"
				},
				"id": 22
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"type": "string",
					"maxlen": 255
				},
				"id": 24
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"type": "string",
					"maxlen": 255
				},
				"id": 25
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"min": 0,
					"max": 86400,
					"scale": 0,
					"step": 1,
					"type": "value"
				},
				"id": 26
			},
			{
				"type": "obj",
				"mode": "wr",
				"property": {
					"type": "string",
					"maxlen": 255
				},
				"id": 27
			},
			{
				"type": "obj",
				"mode": "wr",
				"property": {
					"type": "string",
					"maxlen": 255
				},
				"id": 28
			},
			{
				"mode": "rw",
				"id": 30,
				"type": "raw"
			},
			{
				"mode": "rw",
				"id": 31,
				"type": "raw"
			},
			{
				"mode": "rw",
				"id": 32,
				"type": "raw"
			},
			{
				"mode": "rw",
				"id": 33,
				"type": "raw"
			},
			{
				"type": "obj",
				"mode": "rw",
				"property": {
					"type": "string",
					"maxlen": 255
				},
				"id": 42
			},
			{
				"mode": "rw",
				"id": 55,
				"type": "raw"
			}
		]
	},
	"device_configuration": {
		"Jsonver": "0.0.1",
		"adclimit": 1600,
		"aging": 0,
		"b_lv": 1,
		"b_pin": 24,
		"brightmax": 90,
		"brightmin": 10,
		"c_lv": 1,
		"c_pin": 26,
		"category": "0504",
		"cmod": "rgbc",
		"colormax": 100,
		"colormin": 10,
		"crc": 5,
		"cwtype": 0,
		"defbright": 70,
		"defcolor": "c",
		"deftemp": 100,
		"dmod": 0,
		"g_lv": 1,
		"g_pin": 7,
		"gmkb": 60,
		"gmkg": 60,
		"gmkr": 80,
		"gmwb": 90,
		"gmwg": 90,
		"gmwr": 90,
		"irfunc": 0,
		"ismusic": 1,
		"k1dfunc": 0,
		"k2dfunc": 0,
		"k2lfunc": 0,
		"k2pin_lv": 0,
		"k2pin_pin": 8,
		"k2sfunc": 7,
		"k3dfunc": 0,
		"k3lfunc": 0,
		"k3pin_lv": 0,
		"k3pin_pin": 9,
		"k3sfunc": 12,
		"key_lv": 0,
		"key_pin": 14,
		"keyfunc": 1,
		"knum": 3,
		"ktime": 5,
		"micpin": 23,
		"module": "CBU",
		"notdisturb": 0,
		"onoffmode": 0,
		"pmemory": 1,
		"pwmhz": 1000,
		"r_lv": 1,
		"r_pin": 6,
		"remdmode": 0,
		"rstbr": 30,
		"rstcor": "b",
		"rstmode": 1,
		"rsttemp": 100,
		"sfunc": 3,
		"swgmod": 0,
		"tempstep": 25,
		"title20": 0,
		"wfcfg": "spcl_auto",
		"wfct": 3,
		"wgmod": 0
	},
	"slug": "lsc-3204432-moodlight-rgbw-bulb",
	"image_url": "https://tuya-cloudcutter.github.io/images/thumbs/lsc-3204432-moodlight-rgbw-bulb.jpg"
}