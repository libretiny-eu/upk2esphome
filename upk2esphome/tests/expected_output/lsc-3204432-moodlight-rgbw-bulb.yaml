output:
  - platform: libretiny_pwm
    id: output_red
    pin: P6
  - platform: libretiny_pwm
    id: output_green
    pin: P7
  - platform: libretiny_pwm
    id: output_blue
    pin: P24
  - platform: libretiny_pwm
    id: output_cold
    pin: P26

light:
  - platform: rgbw
    id: light_rgbw
    name: Light
    red: output_red
    green: output_green
    blue: output_blue
    white: output_cold
